nrf_802154/doc/CHANGELOG.rst

```diff
@@ -30,6 +30,10 @@ Removed
 =======
 
 * Removed the :file:`nrf_802154_debug_assert.c` file. (KRKNWK-18116)
+* Removed the deprecated API for :c:func:`nrf_802154_energy_detected`. (KRKNWK-17573)
+  Only the code for ``NRF_802154_ENERGY_DETECTED_VERSION=1`` is kept.
+  Removed API migration macro ``NRF_802154_ENERGY_DETECTED_VERSION``.
```

Comment:
```suggestion
* Removed the deprecated API for the :c:func:`nrf_802154_energy_detected` function. (KRKNWK-17573)
  Only the ``NRF_802154_ENERGY_DETECTED_VERSION=1`` setting can still be used.
  Removed the ``NRF_802154_ENERGY_DETECTED_VERSION`` API migration macro.
```
Please let me know in case the meaning is affected, thanks!

Reply:
The change from
`Only the code for ``NRF_802154_ENERGY_DETECTED_VERSION=1`` is kept.`
to
`Only the ``NRF_802154_ENERGY_DETECTED_VERSION=1`` setting can still be used.`
**changes the meaning**. It cannot be written this way.

The macro `NRF_802154_ENERGY_DETECTED_VERSION` which could have values 0 or 1 and allowed to select API version was removed. The code selected by NRF_802154_ENERGY_DETECTED_VERSION=0 was also removed. What is left is only the code that was selected by NRF_802154_ENERGY_DETECTED_VERSION=1. Because of that the macro NRF_802154_ENERGY_DETECTED_VERSION is no more necessary and it has been removed also.

What about:
```
Removed the code which was selected by ``NRF_802154_ENERGY_DETECTED_VERSION=0`` API migration macro.
Removed the ``NRF_802154_ENERGY_DETECTED_VERSION`` API migration macro itself.
```

Reply:
Got it - thanks for the explanation! Sounds good, then for this part I would just make a small adjustment:

Removed the code selected by the ``NRF_802154_ENERGY_DETECTED_VERSION=0`` API migration macro.
Removed the ``NRF_802154_ENERGY_DETECTED_VERSION`` API migration macro itself.

Reply:
@annwoj, I amended the commit with agreed changes, please have a look.